backend/core/src/main/java/org/sonarsource/sonarlint/core/rules/RuleDetailsAdapter.java

```diff
@@ -81,8 +81,8 @@ public static EffectiveRuleDetailsDto transform(RuleDetails ruleDetails, @Nullab
     return new EffectiveRuleDetailsDto(
       ruleDetails.getKey(),
       ruleDetails.getName(),
-      adapt(ruleDetails.getDefaultSeverity()),
-      adapt(ruleDetails.getType()),
+      ruleDetails.getDefaultSeverity() != null ? adapt(ruleDetails.getDefaultSeverity()) : null,
```

Comment:
You may want to extract the severity/type in a variable to avoid the nullable SL issue